src/QMCSimulation.jl

```diff
@@ -0,0 +1,395 @@
+"""
+    QMCSimulation
+Holds the state and the results of a QMC simulation. Initialise with
+[`init(::FCIQMCProblem)`](@ref) and solve with [`solve!(::QMCSimulation)`].
+
+Obtain the results of a simulation `sm` as a DataFrame with `DataFrame(sm)`.
+"""
+struct QMCSimulation
+    qmc_problem::FCIQMCProblem
+    qmc_state::QMCState
+    report::Report
+    modified::Ref{Bool}
+    aborted::Ref{Bool}
+    success::Ref{Bool}
+    message::Ref{String}
+    elapsed_time::Ref{Float64}
+end
+
+function _set_up_v(dv::AbstractDVec, copy_vectors, _...)
+    # we are not changing type, style or initiator because an AbstractDVec is passed
+    if copy_vectors
+        return deepcopy(dv)
+    else
+        return dv
+    end
+end
+
+function _set_up_v(fdv::FrozenDVec, _, style, initiator, threading)
+    # we are allocating new memory
+    if threading
+        v = PDVec(fdv; style, initiator)
+    elseif initiator isa NonInitiator
+        v = DVec(fdv; style)
+    else
+        v = InitiatorDVec(fdv; style, initiator)
+    end
+    return v
+end
+
+function QMCSimulation(problem::FCIQMCProblem; copy_vectors=true)
+    @unpack hamiltonian, starting_vectors, style, threading, simulation_plan,
+        replica_strategy, shift_strategy, initial_shift_parameters,
+        reporting_strategy, post_step_strategy, time_step_strategy,
+        maxlength, metadata, initiator, random_seed = problem
+
+    n_replicas = num_replicas(replica_strategy)
+
+    # seed the random number generator
+    if !isnothing(random_seed)
+        Random.seed!(random_seed + hash(RMPI.mpi_rank()))
+    end
+
+    # set up the starting vectors and shift parameters
+    if length(starting_vectors) == 1 && n_replicas > 1
+        vectors = ntuple(n_replicas) do i
+            v = _set_up_v(only(starting_vectors), copy_vectors, style, initiator, threading)
+            sizehint!(v, maxlength)
+            return v
+        end
+        shift_parameters = ntuple(n_replicas) do i
+            deepcopy(only(initial_shift_parameters))
+        end
+    else
+        @assert length(starting_vectors) == n_replicas == length(initial_shift_parameters)
+        vectors = map(starting_vectors) do dv
+            v = _set_up_v(dv, copy_vectors, style, initiator, threading)
+            sizehint!(v, maxlength)
+            return v
+        end
+        shift_parameters = deepcopy(initial_shift_parameters)
+    end
+    wm = working_memory(first(vectors))
+
+    # set up the replicas
+    if n_replicas == 1
+        replicas = (ReplicaState(
+            hamiltonian,
+            only(vectors),
+            zerovector(only(vectors)),
+            wm,
+            deepcopy(shift_strategy),
+            deepcopy(time_step_strategy),
+            only(initial_shift_parameters),
```

Comment:
```suggestion
            deepcopy(only(initial_shift_parameters)),
```
These need to be copied. Right now, solving the same problem twice makes the second diverge:

```julia
using Rimu

H = HubbardReal1D(near_uniform(BoseFS{10,10}))

prob = FCIQMCProblem(
    H;
    walltime = 10,
    last_step = 100_000,
)

res1 = solve(prob) # Finishes after 10s
res2 = solve(prob) # Diverges

```

Reply:
Thanks for spotting the bug! This should read `only(shift_parameter)`. They were already copied.